### DISCUSSION TITLE
Release building on Windows does not work for me

### DISCUSSION THREAD

**Discussion by nixxquality** [friendlyanon/cmake-init#110] [Q&A]
I've been having issues building my project, and I'm not sure if something's gone wrong with my environment recently to cause this.

Even with a dead simple repro, I have no luck.

1. I open `Developer Command Prompt for VS 2022`
2. `cd ....../code`
3. `cmake-init will_it_build` with most variables set to default, C++ standard 20, executable, clang-tidy and cppcheck both added to preset.
4. `cd will_it_build`
```
>cmake -S . -B build
-- Building for: Visual Studio 17 2022
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.19045.
-- The CXX compiler identification is MSVC 19.39.33522.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.39.33519/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:31 (find_package):
  By not providing "Findfmt.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "fmt", but
  CMake did not find one.

  Could not find a package configuration file provided by "fmt" with any of
  the following names:

    fmtConfig.cmake
    fmt-config.cmake

  Add the installation prefix of "fmt" to CMAKE_PREFIX_PATH or set "fmt_DIR"
  to a directory containing one of the above files.  If "fmt" provides a
  separate development package or SDK, be sure it has been installed.


-- Configuring incomplete, errors occurred!
```

If you need any more information, please let me know.

Building in dev mode works just fine, with vcpkg installing dependencies and everything. The VCPKG_ROOT environment variable is set and valid.

At the moment, I've settled for building release builds in the dev preset, which doesn't seem correct.

**Comment by friendlyanon** [comment]
You didn't configure the project as instructed. By default, there are `dev` presets you can configure and build with, which you can do as follows:

```
cmake --preset=dev
cmake --build --preset=dev
```

See the generated `HACKING.md` for details.

---

It is also unnecessary to enter a `vcvarsall` virtual environment (e.g. via `Developer Command Prompt for VS 2022`) before doing any of the above, since the default build system is MSBuild, which will enter the virtual environment for its child processes automatically. You would need to do that only if you wanted to use a different generator, such as Ninja.

**Comment by nixxquality** [comment]
Yes, that does build a developer build, but I'm trying to follow the instructions in `BUILDING.md` for a release build.

**Reply by friendlyanon** [reply]
Ah, I misunderstood. Release builds and software distribution in general are a very deep topic, which is why that document only covers the basics and some things to look out for. There really is no one size fits all approach to this, so I don't think I can properly help here.

For example, in [simcity-noinstall](https://github.com/friendlyanon/simcity-noinstall)'s release I sidestep a huge chunk of the potential problems by simply providing a binary that does not depend on the C runtime, but that may not be realistic for vast majority of scenarios.

I don't really have an answer other than you'll have to learn about toolchain (compiler and linker) flags, platform conventions, software distribution methods and the runtime loader for your target platform to make appropriate decisions. A cmake-init generated project will definitely not get in your way and will need no changes specific to this topic to work out.
